src/power_system_simulation/graph_processing.py

```diff
@@ -0,0 +1,193 @@
+
+We define a graph processor class with some function skeletons.
+"""
+
+from typing import List, Tuple
+
+import networkx as nx
+from networkx.exception import NetworkXNoCycle
+
+
+class IDNotFoundError(Exception):
+    """
+    Raising IDNotFoundError exception
+    """
+
+
+class InputLengthDoesNotMatchError(Exception):
+    """
+    Raising InputLengthDoesNotMatchError exception
+    """
+
+
+class IDNotUniqueError(Exception):
+    """
+    Raising IDNotUniqueError exception
+    """
+
+
+class GraphNotFullyConnectedError(Exception):
+    """
+    Raising GraphNotFullyConnectedError exception
+    """
+
+
+class GraphCycleError(Exception):
+    """
+    Raising GraphCycleError exception
+    """
+
+
+class EdgeAlreadyDisabledError(Exception):
+    """
+    Raising EdgeAlreadyDisabledError exception
+    """
+
+
+class GraphProcessor:
+    """
+    General documentation of this class.
+    You need to describe the purpose of this class and the functions in it.
+    We are using an undirected graph in the processor.
+    """
+
+    def __init__(
+        self,
+        vertex_ids: List[int],
+        edge_ids: List[int],
+        edge_vertex_id_pairs: List[Tuple[int, int]],
+        edge_enabled: List[bool],
+        source_vertex_id: int,
+    ) -> None:
+        """
+        Initialize a graph processor object with an undirected graph.
+        Only the edges which are enabled are taken into account.
+        Check if the input is valid and raise exceptions if not.
+        The following conditions should be checked:
+            1. vertex_ids and edge_ids should be unique. (IDNotUniqueError) -- done
+            2. edge_vertex_id_pairs should have the same length as edge_ids. (InputLengthDoesNotMatchError) -- done
+            3. edge_vertex_id_pairs should contain valid vertex ids. (IDNotFoundError) -- done
+            4. edge_enabled should have the same length as edge_ids. (InputLengthDoesNotMatchError) -- done
+            5. source_vertex_id should be a valid vertex id. (IDNotFoundError) -- done
+            6. The graph should be fully connected. (GraphNotFullyConnectedError) -- done
+            7. The graph should not contain cycles. (GraphCycleError) -- done
+        If one certain condition is not satisfied, the error in the parentheses should be raised.
+
+        Args:
+            vertex_ids: list of vertex ids
+            edge_ids: list of edge ids
+            edge_vertex_id_pairs: list of tuples of two integer
+                Each tuple is a vertex id pair of the edge.
+            edge_enabled: list of bools indicating of an edge is enabled or not
+            source_vertex_id: vertex id of the source in the graph
+        """
+        # put your implementation here
+
+        self.graph = nx.Graph()
+        self.graph.add_nodes_from(vertex_ids)
+        self.graph.add_edges_from(
+            [edge_vertex_id_pairs[i] for i in range(len(edge_vertex_id_pairs)) if edge_enabled[i] == 1]
```

Comment:
suggestion: Seeing as it is a list of booleans, you can make the check like this as well! If you want to be explicit, I would suggest using ```edge_enabled[i] == True``` because it more closely matches what is in the list
```suggestion
        self.graph.add_edges_from(
            [edge_vertex_id_pairs[i] for i in range(len(edge_vertex_id_pairs)) if edge_enabled[i]]
```